[INFO   ] Logger: Record log in /storage/emulated/0/Documents/project_utopia/.kivy/logs/kivy_21-09-09_5.txt
[INFO   ] Kivy: v1.11.1
[INFO   ] Kivy: Installed at "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/__init__.py"
[INFO   ] Python: v3.8.3 (default, May 27 2020, 02:08:17) 
[GCC 9.3.0]
[INFO   ] Python: Interpreter at "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/bin/python3"
[INFO   ] Factory: 184 symbols loaded
[INFO   ] Image: Providers: img_tex, img_dds, img_sdl2, img_gif (img_pil, img_ffpyplayer ignored)
[INFO   ] Text: Provider: sdl2
[INFO   ] Window: Provider: sdl2
[INFO   ] GL: Using the "OpenGL ES 2" graphics system
[INFO   ] GL: Backend used <sdl2>
[INFO   ] GL: OpenGL version <b'OpenGL ES 3.2 v1.r26p0-01eac0.###other-sha0123456789ABCDEF0###'>
[INFO   ] GL: OpenGL vendor <b'ARM'>
[INFO   ] GL: OpenGL renderer <b'Mali-G72'>
[INFO   ] GL: OpenGL parsed version: 3, 2
[INFO   ] GL: Texture max size <8192>
[INFO   ] GL: Texture max units <16>
[INFO   ] Window: auto add sdl2 input provider
[INFO   ] Window: virtual keyboard not allowed, single mode, not docked
[WARNING] Base: Unknown <android> provider
[INFO   ] Base: Start application main loop
[INFO   ] GL: NPOT texture support is available
[WARNING] stderr: Traceback (most recent call last):
[WARNING] stderr:   File "kivy/graphics/texture.pyx", line 789, in kivy.graphics.texture.Texture.bind
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/core/text/__init__.py", line 754, in _texture_fill
[WARNING] stderr:     self.render(real=True)
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/core/text/__init__.py", line 695, in render
[WARNING] stderr:     return self._render_real()
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/core/text/__init__.py", line 678, in _render_real
[WARNING] stderr:     self.render_lines(lines, options, self._render_text, y, size)
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/core/text/__init__.py", line 656, in render_lines
[WARNING] stderr:     render_text(line, x, y)
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/core/text/text_sdl2.py", line 54, in _render_text
[WARNING] stderr:     self._surface.render(self, text, x, y)
[WARNING] stderr:   File "kivy/core/text/_text_sdl2.pyx", line 63, in kivy.core.text._text_sdl2._SurfaceContainer.render
[WARNING] stderr: TypeError: an integer is required
[WARNING] stderr: Exception ignored in: 'kivy.graphics.instructions.RenderContext.set_texture'
[WARNING] stderr: Traceback (most recent call last):
[WARNING] stderr:   File "kivy/graphics/texture.pyx", line 789, in kivy.graphics.texture.Texture.bind
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/core/text/__init__.py", line 754, in _texture_fill
[WARNING] stderr:     self.render(real=True)
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/core/text/__init__.py", line 695, in render
[WARNING] stderr:     return self._render_real()
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/core/text/__init__.py", line 678, in _render_real
[WARNING] stderr:     self.render_lines(lines, options, self._render_text, y, size)
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/core/text/__init__.py", line 656, in render_lines
[WARNING] stderr:     render_text(line, x, y)
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/core/text/text_sdl2.py", line 54, in _render_text
[WARNING] stderr:     self._surface.render(self, text, x, y)
[WARNING] stderr:   File "kivy/core/text/_text_sdl2.pyx", line 63, in kivy.core.text._text_sdl2._SurfaceContainer.render
[WARNING] stderr: TypeError: an integer is required
[INFO   ] Base: Leaving application in progress...
[INFO   ] WindowSDL: exiting mainloop and closing.
